compiler/rustc_target/src/target_features.rs

```diff
@@ -328,6 +355,7 @@ const X86_ALLOWED_FEATURES: &[(&str, Stability, ImpliedFeatures)] = &[
     ("sha512", Unstable(sym::sha512_sm_x86), &["avx2"]),
     ("sm3", Unstable(sym::sha512_sm_x86), &["avx"]),
     ("sm4", Unstable(sym::sha512_sm_x86), &["avx2"]),
+    ("soft-float", Forbidden, &[]), // changes float ABI
```

Comment:
ARM targets also have a `soft-float` target feature.  Aarch64 does not, though -- so we can't just add this for all targets, unfortunately.

Reply:
Aarch64 has `abi: "softfloat"`: https://github.com/rust-lang/rust/blob/9b82580c7347f800c2550e6719e4218a60a80b28/compiler/rustc_target/src/spec/targets/aarch64_unknown_none_softfloat.rs#L15

Reply:
That seems like it cannot be changed with `-Ctarget-feature`, which is good news.